client/index.jsx

```diff
@@ -18,174 +18,147 @@ function getDistance(lat1, lon1, lat2, lon2) {
   return d;
 }
 
-class Grass extends React.Component {
-  generateLink() {
-    let grass = this.props.grass;
-    let name = grass.name;
-    let plus = grass.plus_code ? "+" + grass.plus_code.compound_code : "";
+const Grass = ({coords, grass}) => {
+  const { name, plus_code, geometry: { location } } = grass;
+
+  const generateLink = () =>  {
+    const plus = plus_code ? "+" + plus_code.compound_code : "";
     return `https://www.google.com/maps/search/${name}${plus}`;
   }
-  computeDistance() {
-    let srcLoc = this.props.coords;
-    let destLoc = this.props.grass.geometry.location;
+
+  const computeDistance = () => {
+    const srcLoc = coords;
+    const destLoc = location;
     if (!srcLoc || !destLoc) return 0;
-    let lat1 = srcLoc.latitude;
-    let lon1 = srcLoc.longitude;
-    let lat2 = destLoc.lat;
-    let lon2 = destLoc.lng;
+    const {longitude: lon1, latitude: lat1} = srcLoc;
+    const {longitude: lat2, latitude: lon2} = srcLoc;
+
     return getDistance(lat1, lon1, lat2, lon2);
   }
-  render() {
-    let { icon, name } = this.props.grass;
-    return (
-      <a className="p-2 bg-green-500 hover:bg-green-600 rounded-lg w-full"
-         tabIndex="1"
-         href={this.generateLink()}
-         target="_blank">
-        <div className="flex flex-row">
-          <span className="text-left grow">{name}</span>
-          <span className="text-right basis-1/4">({this.computeDistance().toFixed(0)}m)</span>
-        </div>
-      </a>
-    );
-  }
+
+  return (
+    <a className="p-2 bg-green-500 hover:bg-green-600 rounded-lg w-full"
+       tabIndex="1"
+       href={generateLink()}
+       target="_blank">
+      <div className="flex flex-row">
+        <span className="text-left grow">{name}</span>
+        <span className="text-right basis-1/4">({computeDistance().toFixed(0)}m)</span>
+      </div>
+    </a>
+  );
 }
 
-class GrassList extends React.Component {
-  render() {
-    if (this.props.grass.parks.length === 0) {
-      return (
-        <div>
-          <h2 className="p-2 text-2xl">
-            We couldn't find any grass nearby, sorry about that.
-          </h2>
-        </div>
-      );
-    }
+const GrassList = ({coords, grass: { parks } }) => {
+  if (parks?.length === 0) {
```

Comment:
Does Babel desugar this? cursed

Reply:
Yes, it is transformed to something like `parks !== null && parks.length === 0`